### DISCUSSION TITLE
Welcome to reportportal Discussions!

### DISCUSSION THREAD

**Discussion by DzmitryHumianiuk** [reportportal/reportportal#1250] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by bzsharm** [comment]
Hi Team, 

I am part of an R&D team and we are planning to host report portal (shell installation ) in a seperate server. 
Currently we don't have a historical reporting tool , but while doing the shell installation at last we are facing issues with Traefik, as in its not present at all, we followed the exact same steps from the documentation provided in Reportportal.Io

**Reply by DzmitryHumianiuk** [reply]
@bzsharm do you use this? https://github.com/reportportal/shell-installation
and is it relevant to that issue? https://github.com/reportportal/reportportal/issues/1150

**Comment by talmoshe24** [comment]
Hi, 
I have RP 5 integrated with JUnit4 / Maven framework.
All seems to work fine except one issue - when a test type "After Method" fails, I see the fail but it does not fail the test itself. 
Is there a property that enables this functionality? 
I have a class with 20 tests that this "After Method" is called after each test, and it's impossible to identify and categorize which test failed

**Reply by DzmitryHumianiuk** [reply]
@HardNorth can help you here.
@talmoshe24 did you tried latest integration?

**Reply by talmoshe24** [reply]
Hi yes I'm working with RP server 5.3.3 and agent-java-junit 5.0.0-RC-1 (latest)

**Reply by HardNorth** [reply]
@talmoshe24 No, there is no such functionality and I doubt it would be here one day. I personally will be straight against it by numerous reasons. We are reporting tool, we only report what JUnit gives us. We don't fail tests on our own.

There are some workarounds by which you can do this by your own. On your own responsibility.

**Reply by DzmitryHumianiuk** [reply]
@HardNorth do you mean that JUnit4 basically doesn't generate internal events on `AfterMethods`? Events which we can bind over to send data in RP.

**Reply by DzmitryHumianiuk** [reply]
@DzmitryHumianiuk It generates, and JUnit agent sends it as is (as Failed in that case).
The question is about to which exact test of 20 this failure is related and can we (agent) fail related test method if after method failed. 
And I said there is no such functionality.

@talmoshe24 I would say it's more a matter of building the logic correctly in a test framework. 
The test should test. The test should check. The test should make a decision. 
AfterMethod refers to supplementary objects, and its purpose is primarily to clean-up, restore order after the test execution. But do not make a decision on test cases. 
If this is the case, then you should rather think about reworking the approach in the tests.

And this might be the reason why jUnit do not care about sharing the info which `@AfterMethod` belongs to which `@Test`.

**Comment by talmoshe24** [comment]
@DzmitryHumianiuk @HardNorth There are multiple reporting tools (including Jenkins "publish junit test result report") that report a test as failed in case the after test fails.
Since we're moving from debugging the test results from Jenkins to RP, it was raised.

**Reply by DzmitryHumianiuk** [reply]
@talmoshe24 so, the questions is...?

**Comment by dillipkumarpradhan** [comment]
Need urgent help.
Hi,
> I have  integrated RP 5 with my api automation framework (Maven ,Java)
> Have installed RP using DOCKER and also made changes on respective file of my project and when ever I am running my test from my local machine ,It works fine and able to view the real time result on my dashboard which is in local host .

Question
---------
> Now my same API  Automation code is running in GOCD
> How can I integrate the build result from GOCD to report portal UI .

Advance Thanks

**Reply by DzmitryHumianiuk** [reply]
@dillipkumarpradhan make sure your instance of RP (installed locally, as I got from your details) accessible from the GOCD machine.
Just ping it.

and also, double check that your `rp.endpoint` in properties file stated as IP address, but not the `localhost`.

**Reply by dillipkumarpradhan** [reply]
Thanks for the response @DzmitryHumianiuk 

That means I can achieve the things by installing the RP in a ec2 instance and that ec2 instance should be accessible from the GOCD machine and on the rp.endpoint I have to mention the ec2 instance ip address.

**Reply by DzmitryHumianiuk** [reply]
@dillipkumarpradhan yes. Because in properties file you mention for the test agent where to send data, which IP address to use as endpoint.

**Comment by dillipkumarpradhan** [comment]
Hi @DzmitryHumianiuk 
I have deployed RP 5 on a ec2 instance (t3.a medium) .First time when ever I am opening  the portal my cpu utilisation went to 80% and after sometime it goes down to 10% any specific reason behind this ,How can i maintain the cpu utilsation with out increasing the ec2 instance size .

**Reply by DzmitryHumianiuk** [reply]
@dillipkumarpradhan there are list of hidden actions on database size, which will be held based on database schedule.
Actions such as index creation.

In order to use your instance in proportion to your load, we recommend using the following shapes (as in [docs](https://reportportal.io/docs/Optimal-Performance-Hardware)), determined based on performance testing. Use the [documentation](https://reportportal.io/docs/Optimal-Performance-Hardware) to calculate the load model, and determine which size will fit you perfectly.
https://reportportal.io/docs/Optimal-Performance-Hardware

**Comment by AmrWatad** [comment]
Hi @DzmitryHumianiuk 
regarding to pass rate  in v5.1.0
is passing rate per filter on **test level** or on **launches level**
trying to get passing rate per filter, after reading all the version see on 5.1.0 version that's was added
*** Passing rate per filter**
its not clearly if its on test level or launches level

thanx

**Reply by evjlobanova** [reply]
All Passing rates widgets are calculated as Rate = Passed tests/ All tests. The same logic and for launch and for filters

**Comment by dillipkumarpradhan** [comment]
@DzmitryHumianiuk Need one more Help 
If Multiple jobs are triggered From gocd at the same time  and All are pointing to master branch which is connected with report portal server ,Then RP UI is not at all responding , as well as it stops running the test.

Any suggestions to avoid this or rp supports only one build at a single time ,Please confirm

Even if multiple jobs are running ,Is there any configuration that rp can only takes call after finishing the one job

**Reply by evjlobanova** [reply]
Which version of RP do you use? In version 5.4 jobs were rewritten

**Reply by dillipkumarpradhan** [reply]
Thanx for the reply,I am using 5.4.0,Do i need to use 5.5

**Reply by evjlobanova** [reply]
No, you do not need
@irynakozak2 could you please elaborate

**Reply by irynakozak2** [reply]
@dillipkumarpradhan if we're talking about multiple test executions(some pipelines are executed in parallel) and the ReportPortal isn't accessible during that("Loading..." message is displayed or slow UI loading) it means that maybe you need to expand the default connection pool on the `service-API` and also from `PostgreSQL` database side. The possible root cause of the issue is not enough default API connections count for your load profile during the test execution. The default value is "27" and is valid in general for small servers. Please look through these [docs](https://reportportal.io/docs/Optimal-Performance-Hardware%3E6-application-connections-pool-tuning) for tuning connection pool configuration for your instance.
My suggestion is (if you have one `service-API` container, no replication):
service-api
`RP_DATASOURCE_MAXIMUMPOOLSIZE`=100
database:
`max_connections`=500

**Comment by PriyankaSoni1210** [comment]
Hi Team, For POST requests, as per the report Portal API documentation, the response body should only contain - id. However, while testing I encountered that along with id, an auto generated unique id is also a parameter in the response payload. Is the API documentation outdated?

**Reply by DzmitryHumianiuk** [reply]
UUID is the id of element in the message queue.
ID is the id of element in database.

Before saving to database, request goes to the queue, where it will be picked by one of API services for saving. Once it is saved, object will have a physical ID from Database.

**Comment by Haja49** [comment]
Hi Team,
I have been using report portal for last couple of months. But I am facing some issue recently where though test execution completes without any error, launch is not getting added in the report. Please help me here.

Error Message:
Launch 'xxx' not found. Did you use correct Launch ID?

I could observe the same error while debugging the service class (get_launch_info()) of reportportal-client library.
{'errorCode': 4041, 'message': "Launch 'xxx' not found. Did you use correct Launch ID?"}

I am using v5.10.1 for service-api with python pytest-reportportal library.

**Reply by DzmitryHumianiuk** [reply]
Hi @Haja49, 
based on the error message, it seems there's a discrepancy with the Launch ID on the server side, which typically occurs if there's no launch with the specified ID present in the database.

To better assist you, could you provide some additional details about your setup? Specifically, how you're executing the tests and the configuration you have in your parameters file. Any extra context you can give will help us pinpoint the issue more accurately.

**Reply by Haja49** [reply]
@DzmitryHumianiuk 
Below is the docker compose file used for the setup (changed format to .txt as unable to attach as .yml)
[docker-compose.txt](https://github.com/reportportal/reportportal/files/14175118/docker-compose.txt)

OS: ubuntu
Current version:API Service: 5.10.1;Index Service: 5.10.0;Jobs Service: 5.10.0;Authorization Service: 5.10.0;Service UI: 5.10.0;
pytest-reportportal: v5.1.3
reportportal-client: v5.2.5

script is referenced from https://github.com/reportportal/client-Python

Exact error message after suite execution
File "/home/.../lib/python3.10/site-packages/reportportal_client/service.py", line 122, in _get_json
    raise ResponseError(
reportportal_client.errors.ResponseError: **Invalid response: Expecting value: line 1 column 1 (char 0):** <!doctype html><html><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><title>Report Portal</title><meta name="viewport" content="width=device-width,initial-scale=1,minimum-scale=1,maximum-scale=1,user-scalable=no"><meta http-equiv="X-UA-Compatible" content="IE=edge"><link rel="icon" href="favicon.ico" integrity="sha256-aSLjBCQALdnKIlbJ063QGA/8ElrgLRolXWXBRoDyOt4= sha384-Rut7kVmXU0vQgi+BSBPu7ERfVbfylYcbg1w5/+ry42toYCDHwi2pqmQiM5QQanAm" crossorigin="anonymous"><script defer="defer" src="polyfills.app.34483d47.js" integrity="sha256-SswcXJkeow42tR6QILtZzEtUYVGYPXrSwNe8RF0pyVQ= sha384-s7eUXEzN8HiT7+Frz5uKJeogYHNo8CpPqYAuEG9OhVRHVyjXs+PdStTpWEKT8NUZ" crossorigin="anonymous"></script><script defer="defer" src="styles.app.608a049b.js" integrity="sha256-AofOqZL39mcDyCWJJoJHVLVnPa6+hAcoR3tY9cMHAtA= sha384-JagMpHTfndhxOUHGCW/AJzlKzBFfqEB+saHIVdqifosB1sftLx+Q18516kb2ibVk" crossorigin="anonymous"></script><script defer="defer" src="main.app.ffa3d937.js" integrity="sha256-I+IVc5FVn3jHy4XnC5xP8H1Q9UIbEvEARUSCZu0I4rQ= sha384-ZVJgUNyXODJbn2OcqflEt23fjj6oNXhVOa5La58zfSoXyNvn3/akFaiqQWA/yCPw" crossorigin="anonymous"></script><link href="styles.7e311c.css" rel="stylesheet" integrity="sha256-+rFVnWkPai4n11YDFVq1TiLWndacCXYAluxO7pSXnjk= sha384-TuvX/kCl+u5qhA/9w3sJmQLj0v59k7rHw4JtJfZssBB8qvD3VsYvMBP1T3BJIKq/" crossorigin="anonymous"><link href="main.45bb3b.css" rel="stylesheet" integrity="sha256-qUpt/s5eu7MaRfsjY1nkWrLmKCJ9J2pZHND3ZZ7Lg1w= sha384-JBLMQ5s+8AA1IcPr8w8e12QFP1NAvbS+dAMVEhOGYzZkG3djcmNQDXaW8uxA2IFa" crossorigin="anonymous"></head><body><div id="app" style="width: 100%; height: 100%;"></div><div id="popover-root"></div><div id="modal-root"></div><div id="tooltip-root"></div><div id="notification-root"></div><div id="screen-lock-root"></div></body></html>

Since the time of this issue, the service-api container becomes unhealthy frequently. Below are some of the logs from the service-api container.
[service-api-logs.txt](https://github.com/reportportal/reportportal/files/14175187/service-api-logs.txt)

Please note that the issue started happening all of a sudden even though there was no new changes done to the setup and this is blocking me to perform further tests as launches are not getting created.

Please let me know if any other specific info is needed.

**Reply by Haja49** [reply]
@DzmitryHumianiuk Any update on this. We are blocked with this.

**Reply by DzmitryHumianiuk** [reply]
@Haja49, thanks for providing the details. Let's address the key points:

1. The issue stems from a server-side failure, which results in you receiving an HTML page with an error instead of the expected API response. This wouldn't occur if the service were functioning correctly.
![Error illustration](https://github.com/reportportal/reportportal/assets/11332788/1e5a9002-0563-45a8-9f28-4d1a6885e5f7)

2. I've noticed multiple errors in the logs related to "Launch not found." It begins with "Launch 'ecdb1a41-a947-4493-90ae-f9bba0c38daa' not found. Did you use the correct Launch ID?" at line 177, right after API initialization. Additionally, there are no logs indicating the creation of a launch with this ID.

Please ensure that the launch with the UUID (referenced as ID in the logs) mentioned in the error is present in the database's launch table. If you're using a custom integration, make sure to follow the request sequence as outlined in our documentation here: https://reportportal.io/docs/dev-guides/ReportingDevelopersGuide/

Based on my analysis, the situation seems to unfold as follows:
- The launch was not created.
- When test case executions are reported to the API, and the Launch ID (UUID) cannot be found, these reports are sent to the RabbitMQ retry queues.
- Once RabbitMQ reaches its capacity limit, it begins to crash.
- The crashing of RabbitMQ then leads to the API generating an error when new data is reported.
- Consequently, you receive an HTML error response from the server.
- After some time, requests in the RabbitMQ retry queues expire after 10 attempts. RabbitMQ then clears these out and returns to normal operation.


Make sure that the launch with the mentioned ID(uuid) is created.
